pkg/controller/infrastructure/infraflow/shared/whiteboard.go

```diff
@@ -57,6 +59,8 @@ type Whiteboard interface {
 
 	// ImportFromFlatMap reconstructs the hierarchical structure from a flat map containing path-like keys
 	ImportFromFlatMap(data FlatMap)
+	// ImportRoutesFromState imports routes from the state. It is used to import routes from the state into the whiteboard.
+	ImportRoutesFromState(key string, routes []gcp.Route)
```

Comment:
I am not sure why we need this interface method and not use the existing set object. Can you elaborate?

You could pull this out in the reconciler if you wish and the generation increase also does not seem to be useful especially when you just read them from the current state.